openedx/core/djangoapps/course_groups/partition_scheme.py

```diff
@@ -36,16 +40,19 @@ def get_group_for_user(cls, course_key, user, user_partition, track_function=Non
         If the user has no cohort mapping, or there is no (valid) cohort ->
         partition group mapping found, the function returns None.
         """
-        # If the current user is masquerading as being in a group
-        # belonging to the specified user partition, return the
-        # masquerading group or None if the group can't be found.
-        group_id, user_partition_id = get_masquerading_group_info(user, course_key)
-        if user_partition_id == user_partition.id:
-            if group_id is not None:
+        # First, check if we have to deal with masquerading.
+        # If the current user is masquerading as a specific student, use the
+        # same logic as normal to return that student's group. If the current
+        # user is masquerading as a generic student in a specific group, then
+        # return that group.
+        if get_course_masquerade(user, course_key) and not is_masquerading_as_specific_student(user, course_key):
+            group_id, user_partition_id = get_masquerading_group_info(user, course_key)
+            if user_partition_id == user_partition.id and group_id is not None:
```

Comment:
Oops. Sorry for the delay. :+1:
On Jul 23, 2015 2:11 PM, "Braden MacDonald" <notifications@github.com>
wrote:

> In openedx/core/djangoapps/course_groups/partition_scheme.py
> <https://github.com/edx/edx-platform/pull/8995#discussion_r35352760>:
>
> > @@ -36,16 +40,19 @@ def get_group_for_user(cls, course_key, user, user_partition, track_function=Non
> >          If the user has no cohort mapping, or there is no (valid) cohort ->
> >          partition group mapping found, the function returns None.
> >          """
> > -        # If the current user is masquerading as being in a group
> > -        # belonging to the specified user partition, return the
> > -        # masquerading group or None if the group can't be found.
> > -        group_id, user_partition_id = get_masquerading_group_info(user, course_key)
> > -        if user_partition_id == user_partition.id:
> > -            if group_id is not None:
> > +        # First, check if we have to deal with masquerading.
> > +        # If the current user is masquerading as a specific student, use the
> > +        # same logic as normal to return that student's group. If the current
> > +        # user is masquerading as a generic student in a specific group, then
> > +        # return that group.
> > +        if get_course_masquerade(user, course_key) and not is_masquerading_as_specific_student(user, course_key):
> > +            group_id, user_partition_id = get_masquerading_group_info(user, course_key)
> > +            if user_partition_id == user_partition.id and group_id is not None:
>
> @cpennington <https://github.com/cpennington> I agree.
>
> Did you have any further comments about this PR? If not, can I get a +1
> please?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8995/files#r35352760>.
>